plugins/scaffolder-backend-module-gitlab/src/actions/gitlabMergeRequest.ts

```diff
@@ -242,6 +252,23 @@ which uses additional API calls in order to detect whether to 'create', 'update'
         }
       }
 
+      let reviewerIds: number[] = [];
+      if (reviewers !== undefined) {
+        reviewerIds = await Promise.all(
+          reviewers.map(async reviewer => {
+            try {
+              const reviewerUser = await api.Users.all({ username: reviewer });
+              return reviewerUser[0].id;
+            } catch (e) {
+              ctx.logger.warn(
+                `Failed to find gitlab user id for ${reviewer}: ${e}. Proceeding with MR creation without reviewer.`,
+              );
+              return 0; // If set to 0, no reviewer will be set by GitLab
```

Comment:
Should we just filter them using `.filter(Boolean)` and just return `undefined` here or something instead? Seems a little cleaner than letting `gitlab` deal with `0`?

Reply:
Agree, even though the API explicitly mentions 0 to be okay, I wasn't thinking about how it would behave with multiple 0's set when more than one invalid user id is passed.